python/tvm/relay/frontend/tensorflow.py

```diff
@@ -1348,15 +1480,29 @@ def from_tensorflow(self, graph, layout="NHWC", shape=None, outputs=None):
 
             # Infer shapes if passed explicitely
             node_output = self._nodes[node.name]
-            out_type = ir_pass.infer_type(node_output[0])
-            self._output_shapes[node.name] = [get_const_tuple(out_type.checked_type.shape)]
-
+            if shape and (not self._output_shapes[node.name][0]
+                          or -1 in self._output_shapes[node.name][0]):
+                if isinstance(node_output, _expr.TupleWrapper):
+                    out_shapes = []
+                    for tuple_item in node_output:
+                        out_type = ir_pass.infer_type(tuple_item)
+                        out_shapes.append(get_const_tuple(out_type.checked_type.shape))
+                    self._output_shapes[node.name] = out_shapes
+                else:
+                    out_type = ir_pass.infer_type(node_output[0])
+                    self._output_shapes[node.name] = [get_const_tuple(out_type.checked_type.shape)]
```

Comment:
I guess the length of node_output is always 1, no?  If yes, we can create out_shapes in the same way as TupleWrapper.

Reply:
good catch. thanks, I will change